packages/cli/src/cli/index.ts

```diff
@@ -198,6 +198,129 @@ $$$$$$\\   $$$$$$$\\  $$\\  $$$$$$\\   $$$$$$$ |$$\\  $$\\  $$\\  $$$$$$\\  $$$$
       await detectExtensions(options);
     });
 
+  if (!skipIntro) {
+    const versionSpinner = spinner("Checking for updates...");
+    await import("update-notifier").then(
+      async ({ default: updateNotifier }) => {
+        const notifier = updateNotifier({
+          pkg,
+          shouldNotifyInNpmScript: true,
+          // check every time while we're still building the CLI
+          updateCheckInterval: 0,
+        });
+
+        const versionInfo = await notifier.fetchInfo();
+        versionSpinner.stop();
+
+        if (versionInfo.type !== "latest") {
+          const res = await prompts({
+            type: "toggle",
+            name: "upgrade",
+            message: `A new version of the CLI is available. Would you like to upgrade?`,
+            initial: true,
+            active: "yes",
+            inactive: "no",
+          });
+
+          if (res.upgrade) {
+            const updateSpinner = spinner(
+              `Upgrading CLI to version ${versionInfo.latest}...`,
+            );
+
+            const clonedEnvironment = { ...process.env };
+            clonedEnvironment.THIRDWEB_CLI_SKIP_INTRO = "true";
+
+            const installation = await findPackageInstallation();
+
+            // If the package isn't installed anywhere, just defer to npx thirdweb@latest
+            if (!installation) {
+              updateSpinner.succeed(
+                `Now using CLI version ${versionInfo.latest}. Continuing execution...`,
+              );
+
+              await new Promise((done, failed) => {
+                const shell = spawn(
+                  `npx --yes thirdweb@latest ${process.argv
+                    .slice(2)
+                    .join(" ")}`,
+                  [],
+                  { stdio: "inherit", shell: true, env: clonedEnvironment },
+                );
+                shell.on("close", (code) => {
+                  if (code === 0) {
+                    done("");
+                  } else {
+                    failed();
+                  }
+                });
+              });
+
+              return process.exit(0);
+            }
+
+            // Otherwise, get the correct command based on package manager and local vs. global
+            let command = "";
+            switch (installation.packageManager) {
+              case "npm":
+                command = installation.isGlobal
+                  ? `npm install -g thirdweb`
+                  : `npm install thirdweb`;
+                break;
+              case "yarn":
+                command = installation.isGlobal
+                  ? `yarn global add thirdweb`
+                  : `yarn add thirdweb`;
+                break;
+              case "pnpm":
+                command = installation.isGlobal
+                  ? `pnpm add -g thirdweb@latest`
+                  : `pnpm add thirdweb@latest`;
+                break;
+            }
```

Comment:
we need to handle the case here of not being able to detect any supported package manager:
a) technically others exist
b) someone might invent a new one tomorrow

- we should fail gracefully in those cases not just execute an empty string